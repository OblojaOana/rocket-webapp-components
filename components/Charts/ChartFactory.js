import React from "react";
import { ChartLine } from "./components/ChartLine";
import { ChartBar } from "./components/ChartBar";
import { ChartMix } from "./components/ChartMix";
import PropTypes from "prop-types";

export const ChartFactory = ({
  type,
  labels,
  dataInfo,
  underChart,
  chartColor,
  title,
  text,
  StatIcon,
  statText,
  hover,
  displayDataLabels,
  borderWidth,
  options,
  ...other
}) => {
  switch (type) {
    case "Line": {
      return (
        <ChartLine
          labels={labels}
          dataInfo={dataInfo}
          underChart={underChart}
          chartColor={chartColor}
          title={title}
          text={text}
          StatIcon={StatIcon}
          statText={statText}
          hover={hover}
          displayDataLabels={displayDataLabels}
          borderWidth={borderWidth}
          options={options}
          {...other}
        />
      );
    }

    case "Bar": {
      return (
        <ChartBar
          labels={labels}
          dataInfo={dataInfo}
          underChart={underChart}
          chartColor={chartColor}
          title={title}
          text={text}
          StatIcon={StatIcon}
          statText={statText}
          hover={hover}
          displayDataLabels={displayDataLabels}
          options={options}
          {...other}
        />
      );
    }

    case "Mix": {
      return (
        <ChartMix
          dataInfo={dataInfo}
          chartColor={chartColor}
          title={title}
          options={options}
          displayDataLabels={displayDataLabels}
          {...other}
        />
      );
    }

    default:
      return undefined;
  }
};

ChartFactory.propTypes = {
  options: PropTypes.object.isRequired,
  type: PropTypes.oneOf([
    "Line",
    "Bar",
    "Mix"
  ]).isRequired,
  labels: PropTypes.array,
  dataInfo: PropTypes.object.isRequired,
  displayDataLabels: PropTypes.bool.isRequired,
  chartColor: PropTypes.string.isRequired,
  title: PropTypes.string,
  text: PropTypes.string,
  StatIcon: PropTypes.any,
  statText: PropTypes.string,
  hover: PropTypes.bool,
  borderWidth: PropTypes.number,
  underChart: PropTypes.object
};

export default ChartFactory;
