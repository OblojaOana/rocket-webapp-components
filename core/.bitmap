/* THIS IS A BIT-AUTO-GENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. */

{
    "totalsoft_oss.react-mui/new/core@0.0.11": {
        "files": [
            {
                "relativePath": "src/core/index.js",
                "test": false,
                "name": "index.js"
            }
        ],
        "mainFile": "src/core/index.js",
        "trackDir": "src/core",
        "origin": "AUTHORED",
        "exported": true
    },
    "version": "14.8.8"
}