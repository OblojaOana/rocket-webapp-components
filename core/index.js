//Buttons
export { default as Button } from '@bit/totalsoft_oss.react-mui.button'
export { default as Card } from '@bit/totalsoft_oss.react-mui.card'
export { default as AddButton } from '@bit/totalsoft_oss.react-mui.add-button'
export { default as DeleteButton } from '@bit/totalsoft_oss.react-mui.delete-button'
export { default as UploadButton } from '@bit/totalsoft_oss.react-mui.upload-button'
export { default as BackToButton } from '@bit/totalsoft_oss.react-mui.back-to-button'
export { default as IconButton } from '@bit/totalsoft_oss.react-mui.icon-button'
export { default as CancelButton } from '@bit/totalsoft_oss.react-mui.cancel-button'
export { default as DownloadButton } from '@bit/totalsoft_oss.react-mui.download-button'
export { default as DownwardButton } from '@bit/totalsoft_oss.react-mui.downward-button'
export { default as EditButton } from '@bit/totalsoft_oss.react-mui.edit-button'
export { default as SaveButton } from '@bit/totalsoft_oss.react-mui.save-button'
export { default as UpwardButton } from '@bit/totalsoft_oss.react-mui.upward-button'

//Cards
export { default as IconCard } from '@bit/totalsoft_oss.react-mui.icon-card'
export { default as IconCollapseCard } from '@bit/totalsoft_oss.react-mui.icon-collapse-card'
export { default as RegularCard } from '@bit/totalsoft_oss.react-mui.regular-card'
export { default as ChartCard } from '@bit/totalsoft_oss.react-mui.chart-card'
export { default as StatsCard } from '@bit/totalsoft_oss.react-mui.stats-card'
export { default as CardTitle } from '@bit/totalsoft_oss.react-mui.card-title'
export { default as ImageCard } from '@bit/totalsoft_oss.react-mui.image-card'

//FakeText
export { default as LoadingFakeText } from '@bit/totalsoft_oss.react-mui.fake-text'

//Inputs
export { default as CustomTextField } from '@bit/totalsoft_oss.react-mui.custom-text-field'
export { default as DateTime } from '@bit/totalsoft_oss.react-mui.date-time'
export { default as DynamicInput } from '@bit/totalsoft_oss.react-mui.dynamic-input'

//NavPills
export { default as NavPills } from '@bit/totalsoft_oss.react-mui.nav-pills'

//Pagination
export { default as Pagination } from '@bit/totalsoft_oss.react-mui.pagination'

//AutoComplete  
export { default as Autocomplete } from '@bit/totalsoft_oss.react-mui.autocomplete'

//Forbidden and NotFound
export { default as Forbidden } from '@bit/totalsoft_oss.react-mui.forbidden'
export { default as NotFound } from '@bit/totalsoft_oss.react-mui.not-found'

//SimpleSlider
export { default as SimpleSlider } from '@bit/totalsoft_oss.react-mui.simple-slider'

//TimePicker
export { default as TimePicker } from '@bit/totalsoft_oss.react-mui.time-picker'

//ExpandingText
export { default as ExpandingText } from '@bit/totalsoft_oss.react-mui.expanding-text'

//Accordion
export { default as Accordion } from '@bit/totalsoft_oss.react-mui.accordion'
export { default as Piano } from '@bit/totalsoft_oss.react-mui.piano'

//Chart factory
export { default as ChartFactory } from '@bit/totalsoft_oss.react-mui.charts'

//CustomDialog
export { default as CustomDialog } from '@bit/totalsoft_oss.react-mui.custom-dialog'

//CustomLinearProgress
export { default as CustomLinearProgress } from '@bit/totalsoft_oss.react-mui.custom-linear-progress'

//Typography
export { default as Typography } from '@bit/totalsoft_oss.react-mui.typography'

//CheckInternetConnection
export { default as CheckInternetConnection } from '@bit/totalsoft_oss.react-mui.check-internet-connection'

//DialogDisplay
export { default as DialogDisplay } from '@bit/totalsoft_oss.react-mui.dialog-display'

//PasswordField
export { default as PasswordField } from '@bit/totalsoft_oss.react-mui.password-field'

//Toast
export { useToast, ToastContainer } from '@bit/totalsoft_oss.react-mui.toast'