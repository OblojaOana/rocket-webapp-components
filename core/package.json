{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0",
    "url-search-params-polyfill": "^7.0.1"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2",
    "react-i18next": "^10.13.2"
  },
  "sideEffects": false,
  "dependencies": {
    "@bit/totalsoft_oss.react-mui.card": "*",
    "@bit/totalsoft_oss.react-mui.not-found": "*",
    "@bit/totalsoft_oss.react-mui.button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.delete-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.upload-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.back-to-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.icon-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.cancel-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.download-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.downward-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.edit-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.save-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.upward-button": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.icon-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.icon-collapse-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.regular-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.chart-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.stats-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.card-title": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.image-card": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.fake-text": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.custom-text-field": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.date-time": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.dynamic-input": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.nav-pills": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.pagination": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.autocomplete": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.forbidden": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.simple-slider": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.time-picker": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.expanding-text": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.accordion": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.piano": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.charts": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.custom-dialog": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.custom-linear-progress": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.typography": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.check-internet-connection": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.dialog-display": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.password-field": "^0.0.1",
    "@bit/totalsoft_oss.react-mui.toast": "^0.0.1"
  },
  "bit": {
    "env": {
      "compiler": "totalsoft_oss.react-mui/compilers/react@1.0.24"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}