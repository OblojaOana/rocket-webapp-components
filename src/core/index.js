//Buttons
export { default as Button } from './Button'
export { default as AddButton } from './AddButton'
export { default as DeleteButton } from './DeleteButton'
export { default as UploadButton } from './UploadButton'
export { default as BackToButton } from './BackToButton'
export { default as IconButton } from './IconButton'
export { default as CancelButton } from './CancelButton'
export { default as DownloadButton } from './DownloadButton'
export { default as DownwardButton } from './DownwardButton'
export { default as EditButton } from './EditButton'
export { default as SaveButton } from './SaveButton'
export { default as UpwardButton } from './UpwardButton'

//Cards
export { default as IconCard } from './IconCard'
export { default as IconCollapseCard } from './IconCollapseCard'
export { default as RegularCard } from './RegularCard'
export { default as ChartCard } from './ChartCard'
export { default as StatsCard } from './StatsCard'
export { default as CardTitle } from './CardTitle'
export { default as ImageCard } from './ImageCard'

//FakeText
export { default as LoadingFakeText } from './FakeText'

//Inputs
export { default as CustomTextField } from './CustomTextField'
export { default as DateTime } from './DateTime'
export { default as DynamicInput } from './DynamicInput'

//NavPills
export { default as NavPills } from './NavPills'

//Pagination
export { default as Pagination } from './Pagination'

//AutoComplete  
export { default as Autocomplete } from './Autocomplete'

//Forbidden and NotFound
export { default as Forbidden } from './Forbidden/Forbidden'
export { default as NotFound } from './NotFound/NotFound'

//SimpleSlider
export { default as SimpleSlider } from './SimpleSlider'

//TimePicker
export { default as TimePicker } from './TimePicker'

//ExpandingText
export { default as ExpandingText } from './ExpandingText'

//Accordion
export { default as Accordion } from './Accordion'
export { default as Piano } from './Piano'

//Chart factory
export { default as ChartFactory } from './Charts'

export { default as CustomDialog } from './CustomDialog'

export { default as CustomLinearProgress } from './CustomLinearProgress'

export { default as Typography } from './Typography'

export { default as CheckInternetConnection } from './CheckInternetConnection'

export { default as DialogDisplay } from './DialogDisplay'

export { default as PasswordField } from './PasswordField'

//Toast
export { useToast, ToastContainer } from './Toast'