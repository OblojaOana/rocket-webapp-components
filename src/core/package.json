{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2",
    "react-i18next": "^10.13.2"
  },
  "devDependencies": {
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/runtime": "^7.9.6",
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "url-search-params-polyfill": "^7.0.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0"
  },
  "sideEffects": false,
  "dependencies": {
    "react-router-dom": "^5.1.2",
    "chart.js": "2.8.0",
    "i18next": "^17.3.1",
    "i18next-browser-languagedetector": "^3.1.1",
    "i18next-xhr-backend": "^3.2.2",
    "react-datetime": "^2.16.3",
    "@material-ui/styles": "^4.10.0",
    "classnames": "^2.2.6",
    "prop-types": "^15.5.4",
    "lodash.debounce": "^4.0.8",
    "text-mask-addons": "^3.8.0",
    "react-text-mask": "^5.4.3",
    "react-number-format": "^4.4.1",
    "@bit/totalsoft.react-mui.default-logo": "0.0.1",
    "@bit/totalsoft.react-mui.utils.styles": "0.0.1",
    "@date-io/moment": "1.3.13",
    "@material-ui/core": "^4.10.2",
    "@material-ui/icons": "^4.9.1",
    "@material-ui/pickers": "^3.2.10",
    "attr-accept": "^2.2.1",
    "chartjs-plugin-datalabels": "^0.7.0",
    "moment": "^2.27.0",
    "offline-js": "^0.7.19",
    "ramda": "^0.27.1",
    "react-chartjs-2": "^2.9.0",
    "react-select": "^3.1.0",
    "react-toastify": "^6.0.8",
    "use-debounce": "^3.4.3"
  },
  "bit": {
    "env": {
      "compiler": "totalsoft.react-mui/compilers/react@1.0.21"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}
