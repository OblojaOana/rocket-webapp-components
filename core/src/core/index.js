// //Buttons
// export { default as Button } from './Button'
// export { default as AddButton } from './AddButton'
// export { default as DeleteButton } from './DeleteButton'
// export { default as UploadButton } from './UploadButton'
// export { default as BackToButton } from './BackToButton'
// export { default as IconButton } from './IconButton'
// export { default as CancelButton } from './CancelButton'
// export { default as DownloadButton } from './DownloadButton'
// export { default as DownwardButton } from './DownwardButton'
// export { default as EditButton } from './EditButton'
// export { default as SaveButton } from './SaveButton'
// export { default as UpwardButton } from './UpwardButton'

// //Cards
// export { default as IconCard } from './IconCard'
// export { default as IconCollapseCard } from './IconCollapseCard'
// export { default as RegularCard } from './RegularCard'
// export { default as ChartCard } from './ChartCard'
// export { default as StatsCard } from './StatsCard'
// export { default as CardTitle } from './CardTitle'
// export { default as ImageCard } from './ImageCard'

// //FakeText
// export { default as LoadingFakeText } from './FakeText'

// //Inputs
// export { default as CustomTextField } from './CustomTextField'
// export { default as DateTime } from './DateTime'
// export { default as DynamicInput } from './DynamicInput'

// //NavPills
// export { default as NavPills } from './NavPills'

// //Pagination
// export { default as Pagination } from './Pagination'

// //AutoComplete  
// export { default as Autocomplete } from './Autocomplete'

// //Forbidden and NotFound
// export { default as Forbidden } from './Forbidden/Forbidden'
// export { default as NotFound } from './NotFound/NotFound'

// //SimpleSlider
// export { default as SimpleSlider } from './SimpleSlider'

// //TimePicker
// export { default as TimePicker } from './TimePicker'

// //ExpandingText
// export { default as ExpandingText } from './ExpandingText'

// //Accordion
// export { default as Accordion } from './Accordion'
// export { default as Piano } from './Piano'

// //Chart factory
// export { default as ChartFactory } from './Charts'

// //CustomDialog
// export { default as CustomDialog } from './CustomDialog'

// //CustomLinearProgress
// export { default as CustomLinearProgress } from './CustomLinearProgress'

// //Typography
// export { default as Typography } from './Typography'

// //CheckInternetConnection
// export { default as CheckInternetConnection } from './CheckInternetConnection'

// //DialogDisplay
// export { default as DialogDisplay } from './DialogDisplay'

// //PasswordField
// export { default as PasswordField } from './PasswordField'

// //Toast
// export { useToast, ToastContainer } from './Toast'


import LoadingFakeText from '@bit/totalsoft_oss.react-mui.fake-text';
import SaveButton from '@bit/totalsoft_oss.react-mui.save-button';
import RegularCard from '@bit/totalsoft_oss.react-mui.regular-card';
import Piano from '@bit/totalsoft_oss.react-mui.piano';
import Toast from '@bit/totalsoft_oss.react-mui.toast';
import UpwardButton from '@bit/totalsoft_oss.react-mui.upward-button';
import DownwardButton from '@bit/totalsoft_oss.react-mui.downward-button';
import Forbidden from '@bit/totalsoft_oss.react-mui.forbidden';
import Card from '@bit/totalsoft_oss.react-mui.card';
import StatsCard from '@bit/totalsoft_oss.react-mui.stats-card';
import CustomDialog from '@bit/totalsoft_oss.react-mui.custom-dialog';
import UploadButton from '@bit/totalsoft_oss.react-mui.upload-button';
import Pagination from '@bit/totalsoft_oss.react-mui.pagination';
import TimePicker from '@bit/totalsoft_oss.react-mui.time-picker';
import PasswordField from '@bit/totalsoft_oss.react-mui.password-field';
import FileUploadButton from '@bit/totalsoft_oss.react-mui.file-upload-button';
import DynamicInput from '@bit/totalsoft_oss.react-mui.dynamic-input';
import Charts from '@bit/totalsoft_oss.react-mui.charts';
import NavPills from '@bit/totalsoft_oss.react-mui.nav-pills';
import DeleteButton from '@bit/totalsoft_oss.react-mui.delete-button';
import CustomLinearProgress from '@bit/totalsoft_oss.react-mui.custom-linear-progress';
import IconButton from '@bit/totalsoft_oss.react-mui.icon-button';
import CheckInternetConnection from '@bit/totalsoft_oss.react-mui.check-internet-connection';
import SimpleSlider from '@bit/totalsoft_oss.react-mui.simple-slider';
import ImageCard from '@bit/totalsoft_oss.react-mui.image-card';
import Button from '@bit/totalsoft_oss.react-mui.button';
import CustomTextField from '@bit/totalsoft_oss.react-mui.custom-text-field';
import BackToButton from '@bit/totalsoft_oss.react-mui.back-to-button';
import DateTime from '@bit/totalsoft_oss.react-mui.date-time';
import IconCollapseCard from '@bit/totalsoft_oss.react-mui.icon-collapse-card';
import DialogDisplay from '@bit/totalsoft_oss.react-mui.dialog-display';
import NotFound from '@bit/totalsoft_oss.react-mui.not-found';
import Typography from '@bit/totalsoft_oss.react-mui.typography';
import IconCard from '@bit/totalsoft_oss.react-mui.icon-card';
import EditButton from '@bit/totalsoft_oss.react-mui.edit-button';
import CardTitle from '@bit/totalsoft_oss.react-mui.card-title';
import CancelButton from '@bit/totalsoft_oss.react-mui.cancel-button';
import DownloadButton from '@bit/totalsoft_oss.react-mui.download-button';

export default { LoadingFakeText, SaveButton, RegularCard }