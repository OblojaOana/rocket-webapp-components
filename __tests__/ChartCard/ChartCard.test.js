import React from 'react';
import renderer from 'react-test-renderer';
import ChartFactory from '../../components/Charts/ChartFactory';
import AccessTime from "@material-ui/icons/AccessTime";
import { Grid } from '@material-ui/core';

jest.mock('react-chartjs-2', () => ({
    Bar: () => null, // add any additional chart types here
    Line: () => null
}));

jest.mock('chartjs-plugin-datalabels', () => ({
    Bar: () => null, // add any additional chart types here
    Line: () => null
}));

const labels = ["January", "February", "March", "April", "May", "June",
    "July", "August", "September", "October", "November", "December"]

const series = [
    [12, 28, 38, 5, 8, 13, 11, 22, 44, 103, 99, 27],
    [221, 240, 445, 474, 324, 180, 117, 137, 277, 801, 508, 114]
]

const options = {
    maintainAspectRatio: false,
    title: {
        display: false,
        text: ""
    },
    layout: {
        padding: {
            left: 0,
            right: 20,
            top: 20,
            bottom: 0
        }
    },
    legend: {
        display: false,
        position: "bottom"
    },
    scales: {
        xAxes: [
            {
                offset: false,
                gridLines: {
                    display: true,
                    color: "white",
                    borderDash: [1, 5],
                    zeroLineColor: "rgb(255,255,255)",
                    drawBorder: false,
                    zeroLineWidth: 0
                },
                scaleLabel: {
                    display: false,
                    labelString: "",
                    fontColor: "black"
                },
                ticks: {
                    fontColor: "white",
                    autoSkip: false
                },
                barPercentage: 0.2
            }
        ],
        yAxes: [
            {
                gridLines: {
                    display: true,
                    color: "white",
                    borderDash: [1, 5],
                    zeroLineColor: "rgb(255,255,255)",
                    drawBorder: false,
                    zeroLineBorderDash: [1, 5]
                },
                scaleLabel: {
                    display: false,
                    labelString: "",
                    fontColor: "black"
                },
                ticks: {
                    fontColor: "white"
                }
            }
        ]
    }
};


describe('ChartCard', () => {
    it('snapshot', () => {
        const component = renderer.create(<Grid container spacing={1}>
            <Grid item xs={12} sm={12} md={12} lg={12}>
                <ChartFactory
                    type="Line"
                    labels={labels}
                    dataInfo={{
                        datasets: [
                            {
                                data: series[0],
                                color: "white",
                                label: "Approved"
                            },

                            {
                                data: series[1],
                                color: "yellow",
                                label: "In work"
                            }
                        ]
                    }}
                    underChart={
                        <div>
                            Some content under chart
                            </div>
                    }
                    chartColor={"blue"}
                    StatIcon={AccessTime}
                    statText={'Text description'}
                    title={'Title'}
                    hover={true}
                    displayDataLabels={false}
                    borderWidth={5}
                    options={options}
                    fill={false}
                />
            </Grid>

            <Grid item xs={12} sm={12} md={12} lg={12}>
                <ChartFactory
                    type="Bar"
                    labels={labels}
                    dataInfo={{
                        datasets: [
                            {
                                data: series[0],
                                color: "white"
                            }
                        ]
                    }}
                    underChart={
                        <div>
                            Some content under chart
                            </div>
                    }
                    chartColor={"orange"}
                    title={'Title'}
                    StatIcon={AccessTime}
                    statText={'Status text'}
                    hover={true}
                    displayDataLabels={false}
                    options={options}
                />
            </Grid>
        </Grid>);

        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});