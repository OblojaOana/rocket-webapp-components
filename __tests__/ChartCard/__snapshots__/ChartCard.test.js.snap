// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartCard snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MuiCardHeader-root makeStyles-cardHeader-2 makeStyles-blueCardHeader-6"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          />
        </div>
      </div>
      <div
        className="MuiCardContent-root makeStyles-cardContent-9"
      >
        <div
          className="makeStyles-underChart-23"
        >
          <div>
            Some content under chart
          </div>
        </div>
        <h6
          className="MuiTypography-root makeStyles-cardTitle-10 MuiTypography-subtitle1"
        >
          Title
        </h6>
        <p
          className="MuiTypography-root makeStyles-cardCategory-11"
        />
      </div>
      <div
        className="MuiCardActions-root makeStyles-cardActions-12 MuiCardActions-spacing"
      >
        <div
          className="makeStyles-cardStats-13"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-cardStatsIcon-14 makeStyles-grayCardStatsIcon-21"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          Text description
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
  >
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="MuiCardHeader-root makeStyles-cardHeader-2 makeStyles-orangeCardHeader-3"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          />
        </div>
      </div>
      <div
        className="MuiCardContent-root makeStyles-cardContent-9"
      >
        <div
          className="makeStyles-underChart-23"
        >
          <div>
            Some content under chart
          </div>
        </div>
        <h6
          className="MuiTypography-root makeStyles-cardTitle-10 MuiTypography-subtitle1"
        >
          Title
        </h6>
        <p
          className="MuiTypography-root makeStyles-cardCategory-11"
        />
      </div>
      <div
        className="MuiCardActions-root makeStyles-cardActions-12 MuiCardActions-spacing"
      >
        <div
          className="makeStyles-cardStats-13"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-cardStatsIcon-14 makeStyles-grayCardStatsIcon-21"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          Status text
        </div>
      </div>
    </div>
  </div>
</div>
`;
